interactions.go

```diff
@@ -7,8 +7,230 @@ import (
+	GuildID   string                            `json:"guild_id"`
+	ChannelID string                            `json:"channel_id"`
+	Member    *Member                           `json:"member"`
+	Token     string                            `json:"token"`
+	Version   int                               `json:"version"`
+}
+
+// ApplicationCommandInteractionData is representing interaction data for application command.
+type ApplicationCommandInteractionData struct {
+	ID      string                                     `json:"id"`
+	Name    string                                     `json:"name"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options"`
+}
+
+// ApplicationCommandInteractionDataOption is representing an option of application's command.
+type ApplicationCommandInteractionDataOption struct {
+	Name string `json:"name"`
+	// Contains the value specified by InteractionType
+	Value   interface{}                                `json:"value,omitempty"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options,omitempty"`
+}
+
+// IntValue is utility function for casting option value to integer
+func (o ApplicationCommandInteractionDataOption) IntValue() (i int64) {
+	if v, ok := o.Value.(float64); ok {
+		i = int64(v)
+	}
+	return
+}
+
+// UintValue is utility function for casting option value to unsigned integer
+func (o ApplicationCommandInteractionDataOption) UintValue() (i uint64) {
+	if v, ok := o.Value.(float64); ok {
+		i = uint64(v)
+	}
+	return
+}
+
+// FloatValue is utility function for casting option value to float
+func (o ApplicationCommandInteractionDataOption) FloatValue() (n float64) {
+	if v, ok := o.Value.(float64); ok {
+		n = float64(v)
+	}
+	return
+}
+
+// StringValue is utility function for casting option value to string
+func (o ApplicationCommandInteractionDataOption) StringValue() (s string) {
+	if v, ok := o.Value.(string); ok {
+		s = v
+	}
+	return
+}
+
+// BoolValue is utility function for casting option value to bool
+func (o ApplicationCommandInteractionDataOption) BoolValue() (b bool) {
+	if v, ok := o.Value.(bool); ok {
+		b = v
+	}
+	return
+}
+
+// ChannelValue is utility function for casting option value to channel object.
+// s : Session object, if not nil, function additionaly fetches all channel's data
+func (o ApplicationCommandInteractionDataOption) ChannelValue(s *Session) (ch *Channel) {
+	chanID := o.StringValue()
+	if chanID == "" {
+		return
+	}
+
+	if s == nil {
+		return &Channel{ID: chanID}
+	}
+
+	ch, err := s.State.Channel(chanID)
+	if err != nil {
+		ch, err = s.Channel(chanID)
+	}
+
+	return
+}
+
+// RoleValue is utility function for casting option value to role object.
+// s : Session object, if not nil, function additionaly fetches all role's data
+func (o ApplicationCommandInteractionDataOption) RoleValue(s *Session, gID string) (r *Role) {
+	roleID := o.StringValue()
+	if roleID == "" {
+		return
+	}
+
+	if s == nil || gID == "" {
+		return &Role{ID: roleID}
```

Comment:
why not 
```
r = &Role{ID: roleID}
return
```

Please stop using named variable in response in all places. I don't see problem use `return &Role{ID: roleID}` in all places.

```
var err error
r, err = s.State.Role....

r, err := s.State.Role
return r
```

second variant more readable

Reply:
I don't see any **reasonable** point to change this.

Reply:
It is recommended by the Go team to use named returns sparingly, notably they say `Naked return statements should be used only in short functions... . They can harm readability in longer functions.`. 
This is not a particularly short function, and its a bit awkward to be seeing empty returns here. In fact, it made my reading this function in this review a bit confusing.
Please use explicit returns here and not named returns :slightly_smiling_face:

Reply:
As I can see, I already solved this problem, I'm going to close this conversation for now.